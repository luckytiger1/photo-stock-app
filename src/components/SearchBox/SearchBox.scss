@font-face {
  font-family: 'SF UI Display Light';
  font-style: normal;
  font-weight: normal;
  src: local('SF UI Display Light'),
    url('sf-ui-display-light-58646b33e0551.woff') format('woff');
}

.search-box {
  background: black;
  margin-bottom: 70px;
}

.input-field {
  border-bottom: 1px solid;
  border-image: linear-gradient(
      to left,
      #000000 0%,
      rgba(224, 218, 218, 0.747) 50%,
      #000000 100%
    )
    0 0 100% 0/0 0 1px 0 stretch;
  margin-bottom: 30px;
}

#search-input {
  border: none;
  max-width: 700px;
  min-width: 350px;
  text-align: center;
  padding: 0 15px;
  margin-bottom: 4px;
  font-family: 'SF UI Display Light';
  font-size: 2rem;
  background: black;
  line-height: 60px;
  color: white;
  &:focus {
    outline: none;
  }
  &::placeholder {
    text-align: center;
    color: white;
  }
}
