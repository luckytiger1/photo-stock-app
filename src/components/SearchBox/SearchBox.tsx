import React from 'react';
import HistoryListContainer from '../HistoryList/HistoryListContainer';

const SearchBox = ({ handleEnterKey }: any) => {
  return (
    <div className="search-box">
      <div className="input-field text-center ">
        <input
          type="text"
          placeholder="Search"
          id="search-input"
          onKeyDown={handleEnterKey}
        />
      </div>
      <HistoryListContainer />
    </div>
  );
};

export default SearchBox;
