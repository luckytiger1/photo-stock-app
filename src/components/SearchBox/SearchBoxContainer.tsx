import React from 'react';
import { connect } from 'react-redux';
import { setTerm } from '../../redux/actions/fetchImages';
import { resetPage } from '../../redux/actions/advancePage';
import './SearchBox.scss';
import SearchBox from './SearchBox';

interface SearchBoxContainerProps {
  handleSearch: (data: string) => void;
  resetPageCount: () => void;
}

const SearchBoxContainer: React.FC<SearchBoxContainerProps> = ({
  handleSearch,
  resetPageCount,
}) => {
  const handleEnterKey = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      resetPageCount();
      handleSearch((e.target as HTMLInputElement).value);
    }
  };
  return <SearchBox handleEnterKey={handleEnterKey} />;
};

const mapDispatchToProps = {
  handleSearch: setTerm,
  resetPageCount: resetPage,
};

export default connect(null, mapDispatchToProps)(SearchBoxContainer);
