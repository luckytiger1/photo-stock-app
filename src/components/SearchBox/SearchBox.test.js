import React from 'react';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import SearchBox from './SearchBox';

it('<SearchBox /> renders correctly', () => {
  expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
  const mockEnterKey = jest.fn();
  expect(
    shallow(<SearchBox handleEnterKey={mockEnterKey} />),
  ).toMatchSnapshot();
});
