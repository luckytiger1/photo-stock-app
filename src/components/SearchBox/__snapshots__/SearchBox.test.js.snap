// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> renders correctly 1`] = `
<div
  className="search-box"
>
  <div
    className="input-field text-center "
  >
    <input
      id="search-input"
      onKeyDown={[MockFunction]}
      placeholder="Search"
      type="text"
    />
  </div>
  <withRouter(Connect(HistoryListContainer)) />
</div>
`;
